bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java

```diff
@@ -1712,4 +1716,74 @@ public void setBookieAuthProviderFactoryClass(String factoryClass) {
     public String getBookieAuthProviderFactoryClass() {
         return getString(BOOKIE_AUTH_PROVIDER_FACTORY_CLASS, null);
     }
+
+    /**
+      * Whether SSL is enabled on the bookie. Bookies use the same port for SSL
+      * and non-SSL traffic, using a STARTTLS mechanism to initiate an SSL handshake.
+      * If SSL is enabled, then a keystore containing a cert must be configured.
+      * @see #setSSLKeyStore(String)
+      * @see #setSSLKeyStorePassword(String)
+      * @param Whether to enable SSL or not
+      * @return the server configuration
+      */
+     public ServerConfiguration setSSLEnabled(boolean enabled) {
+         setProperty(SSL_ENABLED, enabled);
+         return this;
+     }
+
+     /**
+      * Whether SSL is enabled on the bookie.
+      * @see #setSSLEnabled(boolean)
+      * @return whether SSL is enabled on the bookie or not.
+      */
+     public boolean isSSLEnabled() {
+         return getBoolean(SSL_ENABLED, false);
+     }
+
+     /**
+      * Set the keystore containing the SSL certificate to be used during SSL
+      * negotiation. The keystore must be in PKCS12 format. The keystore can be
+      * generated using keytool, which comes as standard with the JDK.
+      *
+      * The path specified can be a filesystem path or a resource path. In the
+      * case of a resource path, the keystore must exist in the classpath. The
+      * bookie will interpret the path as a resource path first, and if the
+      * keystore cannot be found on the classpath, it will try to interpret it
+      * as a filesystem path.
+      *
+      * @param keyStore Path to the keystore
+      * @return the server configuration
+      */
+     public ServerConfiguration setSSLKeyStore(String keyStore) {
+         setProperty(SSL_KEY_STORE, keyStore);
+         return this;
+     }
+
+     /**
+      * Get the configured keystore, containing the certificate to be used for
+      * SSL negotiation.
+      *
+      * @return the path to the keystore. This may be a filesystem or a resource path.
+      */
+     public String getSSLKeyStore() {
+         return getString(SSL_KEY_STORE, "nonexistent_keystore");
+     }
+
+     /**
+      * Set the password to decrypt the configured SSL keystore.
+      * @see #setSSLKeyStore(String)
+      * @return the server configuration
+      */
+     public ServerConfiguration setSSLKeyStorePassword(String keyStorePassword) {
```

Comment:
I have no strong opinion on this. Usually other common libraries use properties to store directly passwords and no path to files.
If you need a path to file I can change it.

b.q.
will the access to an absolute file system path  lead to other security concerns ?